lib/lockfile_test.go

```diff
@@ -0,0 +1,30 @@
+package lib
+
+import (
+	"runtime"
+	"testing"
+)
+
+// Test Locking
+func TestLocking(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Log("Skipping test on Windows: cannot figure it out 01-Mar-2025")
+	} else {
+		var lockFile string = "../test-data/lockfile.lock"
+
+		// Acquire lock
+		if lockAcquireErr := acquireLock(lockFile, 1, 1); lockAcquireErr == nil {
```

Comment:
I think the main tests I think would be needed are:
 * Test running aquireLock
   *  Ensure lock file is created
 * Test create the lock file and then releaseLock:
   *  ensure the file is removed
 * Test running aquireLock and then running aquireLock
   * ensuring that an error is returned

Reply:
Working on it. Thanks for great review comments! Appreciate it 🤝

Reply:
Hope I got this working as expected 🤞🏻 c3ef2e3